webapp/graphite/render/functions.py

```diff
@@ -2958,6 +2957,33 @@ def reduceSeries(requestContext, seriesLists, reduceFunction, reduceNode, *reduc
     metaSeries[key].name = key
   return [ metaSeries[key] for key in keys ]
 
+def applyByNode(requestContext, seriesList, nodeNum, templateFunction, newName=None):
+  """
+  Takes a seriesList and applies some complicated function (described by a string), replacing templates with uniquified
+  values from the seriesList up to some unique node.
+
+  ...code-block:: none
+
+      &target=applyByNode(servers.*.disk.bytes_free,1,"divideSeries(%.disk.bytes_free,sumSeries(%.disk.bytes_*))")
+
+  Would find all series which match `servers.*.disk.bytes_free`, then trim them down to unique series up to the node
+  given by nodeNum, then fill them into the template function provided (replacing % by the prefixes)
+  """
```

Comment:
First, a simple nit - this sentence is missing a period at the end.

I'm pretty sure I grok the intent of this function, but perhaps you could elaborate by giving a series of intermediate examples demonstrating how each series would be interpolated into the `templateFunction`?